modules/powers/Nemesis.class.php

```diff
@@ -15,7 +15,148 @@ public function __construct($game, $playerId)
     $this->playerCount = [2, 3, 4];
     $this->golden  = false;
     $this->orderAid = 42;
+
+    $this->implemented = true;
+  }
+
+
+  public function stateAfterBuild()
+  {
+    $arg = [];
+    $this->argUsePower($arg);
+    return (count($arg['workers']) > 0)? 'power' : null;
+  }
+
+
+  public function argUsePower(&$arg)
+  {
+    $arg['power'] = $this->id;
+    $arg['power_name'] = $this->name;
+    $arg['skippable'] = true;
+    $arg['workers'] = $this->game->board->getPlacedActiveWorkers();
+    foreach($arg['workers'] as &$worker)
+    {
+      $worker['works'] = [];
+    }
+	$workers = $this->game->board->getPlacedActiveWorkers();
+	if (count($workers) != 2)
+	  throw new BgaVisibleSystemException('Unexpected state in Nemesis.');
+
+	$actions = $this->game->log->getLastWorks(['force'], null, 2);
+
+	# choose second exchange
+	if (count($actions) > 0)
```

Comment:
I don't think the behavior needs to be any different, whether it is the first or second exchange? We need to maintain a list of IDs for previously-exchanged workers and filter those out, but otherwise the rest of behavior is the same, correct?

Reply:
in a 3-player game, you should be able to target only teammates of the previously targetted opponent worker. I agree there should be a better way to factorize some lines here, but there is a "big" difference.

Reply:
Also, one pair of workers has already been swapped when dealing with the second so it is not possible to do the same test of adjacency.

Reply:
Then instead, compute the possible works in stateAfterBuild and save that information in the log as special action like usePowerNemesis?

Reply:
I don't understand what is wrong here, I'm actually not sure we can factorize much here.